src/target/source/codegen_c_host.cc

```diff
@@ -57,6 +60,48 @@ void CodeGenCHost::AddFunction(const PrimFunc& f) {
   CodeGenC::AddFunction(f);
 }
 
+void CodeGenCHost::LinkParameters(Map<String, LinkedParam> params) {
+  PrintFuncPrefix();
+  stream << " " << tvm::runtime::symbol::tvm_lookup_linked_param
+         << "(void* args, int* arg_type_ids, int num_args, void* out_ret_value, "
+         << "int* out_ret_tcode, void* resource_handle) {\n";
+  ICHECK_EQ(GetUniqueName(tvm::runtime::symbol::tvm_lookup_linked_param),
+            tvm::runtime::symbol::tvm_lookup_linked_param)
+      << "builtin PackedFunc name already taken: " << tvm::runtime::symbol::tvm_lookup_linked_param;
+  stream << "    switch (((int64_t*) args)[0]) {\n"
+         << "    default:\n"
+         << "        out_ret_tcode[0] = " << kTVMNullptr << ";\n"
+         << "        return 0;\n";
+
+  function_names_.emplace_back(tvm::runtime::symbol::tvm_lookup_linked_param);
+  for (auto kv : params) {
+    decl_stream << "\n"
+                << "#ifdef __cplusplus\n"
+                << "extern \"C\" {\n"
+                << "#endif\n"
+                << "static const ";
+    int64_t num_elements = 1;
+    for (int64_t dim : kv.second->param.Shape()) {
+      num_elements *= dim;
+    }
+    PrintType(kv.second->param.DataType(), decl_stream);
+    decl_stream << " " << ::tvm::runtime::symbol::tvm_param_prefix << kv.first << "["
+                << num_elements << "] = {\n";
+    NDArrayDataToC(kv.second->param, 4, decl_stream);
+    decl_stream << "};\n"
+                << "#ifdef __cplusplus\n"
+                << "}  // extern \"C\"\n"
+                << "#endif\n";
+    stream << "    case " << kv.second->id << ":\n"
+           << "        ((uint64_t*)out_ret_value)[0] = (uint64_t) (uintptr_t) "
```

Comment:
it's just copying the pointer here. I believe that should be ok?

Reply:
Oh ok. So kv,first is the array's name, right ? Any reason why is it casted to uint64_t ? Im thinking of a non 64 bit platform usecase.

Reply:
correct. ordinarily we would cast to void*, but since void* may be less than 64 bits, we cast to uint64_t which is the largest type stored in TVMValue.

Reply:
I still think void* should be enough here. void* would be less than 64 bits if the machine compiled for is less than 64bits. right? otherwise this will make it store the pointer in 64bit space where the machine could have a address space defined less than 64 bits, unless Im missing something here. In this specific case, the array defined for the constant will have an address depending the target machine it would be compiled for, hence my confusion.

Do we need to perform pointer arithmetic on this ? -- Im asking this because uintptr_t cast is used. If so, using uintptr_t should be sufficient instead of void*.

Reply:
TVMValue is always 64 bits wide, though, because it's a union including int64_t. so when addressing it as an array, need to cast appropriately. i'm not sure if the uintptr_t cast is excessive--quite possibly it is.

Reply:
Alright then. I will take your word for it. :). See if you can drop uintptr_t if that is not required given that it is going to be written to 64bit wide space location anyway.